Source/DataSources/GroundGeometryUpdater.js

```diff
@@ -120,21 +132,32 @@ define([
     /**
      * @private
      */
-    GroundGeometryUpdater.getGeometryHeight = function(heightProperty, heightReferenceProperty, time) {
-        var heightReference = Property.getValueOrDefault(heightReferenceProperty, time, HeightReference.NONE);
+    GroundGeometryUpdater.getGeometryHeight = function(height, heightReference) {
+        if (!defined(height)) {
+            if (heightReference !== HeightReference.NONE) {
```

Comment:
Can't `heightReference` also be `undefined`?  If not, then maybe add a `Check` call for it behind a debug pragma?  I know this is a private function, but it looks like you've moved the responsibility to defaulting `HeightReference.NONE` up to the caller.

Reply:
It should be defaultValued to `NONE` before it gets here, so yeah I'll add a `Check`